custom-metrics-stackdriver-adapter/pkg/adapter/translator/query_builder.go

```diff
@@ -91,6 +91,80 @@ type Translator struct {
 	supportDistributions bool
 }
 
+// podValues is a helper struct to hold pods values
+type podValues struct {
+	pods     *v1.PodList
+	podNames []string
+}
+
+// isPodsContainerValid checks if podValues is valid to be used
+//
+// this only happens when the both pods & podNames are provided,
+// for example, when WithPods() and WithPodNames() are both used in QueryBuilder
+func (pc podValues) isPodsContainerValid() bool {
+	return !(len(pc.podNames) > 0 && pc.pods != nil && len(pc.pods.Items) > 0)
+}
+
+// isPodsEmpty checks if pods container is empty
+func (pc podValues) isPodsContainerEmpty() bool {
+	return len(pc.podNames) == 0 && (pc.pods == nil || len(pc.pods.Items) == 0)
+}
+
```

Comment:
Should we also update the function name since the struct name is updated?

Reply:
If you want to comment a function, please keep the name consistency. It uses `isPodsEmpty` in comment but the function name is isPodsContainerEmpty.